### PR TITLE
NISP-1618: Post SPa exclusion

### PR DESCRIPTION
**Pull request by amyhart6**
* added exclusion for people with new State Pension. 
* they are excluded from State Pension age - 1 day onwards
* fixed bug where 'now' was constant time
* https://github.com/hmrc/nisp-frontend/pull/41 needs to be merged first
* With @dspork

### CODE REVIEW

File: test/uk/gov/hmrc/nisp/services/SPExclusionsServiceSpec.scala [hunk 1]
```diff
@@ -39,135 +38,161 @@ class SPExclusionsServiceSpec extends UnitSpec with OneAppPerSuite  {
...
       }
 
       "return dissonance exclusion for 199 provided and 200 calculated" in {
-        SPExclusionsService(30, countryGB, false, "M", NpsDate(1959, 1, 1), List(), None, nino, List(), 199, 200)
+        SPExclusionsService(30, countryGB, false, "M", List(), None, nino, List(), 199, 200, defaultDate, statePensionAge)
           .getSPExclusions shouldBe createModelWithListItems(SPExclusion.AmountDissonance)
       }
     }
+
+    "checking if they have reached State Pension age -1 day on or after 6 April 2016" should {
```
Comment (andrewgee) [inline test/uk/gov/hmrc/nisp/services/SPExclusionsServiceSpec.scala]: I think the assertions should be in separate tests so it's easier to work out what has failed.

**Comment by dspork** [comment]
@andrewgee I've addressed your comments

File: test/uk/gov/hmrc/nisp/services/SPExclusionsServiceSpec.scala [hunk 2]
```diff
@@ -39,135 +38,184 @@ class SPExclusionsServiceSpec extends UnitSpec with OneAppPerSuite  {
...
       "return dissonance exclusion for 199 provided and 200 calculated" in {
-        SPExclusionsService(30, countryGB, false, "M", NpsDate(1959, 1, 1), List(), None, nino, List(), 199, 200)
+        SPExclusionsService(30, countryGB, false, "M", List(), None, nino, List(), 199, 200, defaultDate, statePensionAge)
           .getSPExclusions shouldBe createModelWithListItems(SPExclusion.AmountDissonance)
       }
     }
-  }
 
+    "checking if they have reached State Pension age -1 day on or after 6 April 2016" should {
+      "return no exclusions for pre SPa -1 and on or after 6 April 2016" should {
```
Comment (andrewgee) [inline test/uk/gov/hmrc/nisp/services/SPExclusionsServiceSpec.scala]: should this `should` be `when`?

**Comment by dspork** [comment]
@andrewgee fixed

**Comment by andrewgee** [comment]
All good!

**Pull request merged**